### PR TITLE
Parkinfo object query

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This should ultimately be for #15968. 
~~Very much a WiP.~~

~~At present bpb would output https://gist.github.com/duncanspumpkin/268ed21112f9715d8e9cdb3434659d64
Will probably have to create a new mock ObjectRepo class to be able to output the objects to the desired location.~~

~~Notes: Context required to be able to import s4 and s6. So if export of objects wanted need to create a mock Context and mock ObjectRepo. Need to be able to load the whole object to be able to verify if object is official or custom as json identifier's are not storing this info.~~

I've decided that making it unpack objects to a specified directory is too awkward to do atm so instead this will just output all the in use objects and their identifier. This is useful for people wanting to check what objects are in use in a park.

This is a typical output from the tool.
https://gist.github.com/duncanspumpkin/2694cbaff93af31ba5e9400dc71ba17c

### CODE REVIEW

File: src/openrct2/object/ObjectList.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ int32_t object_entry_group_encoding[] = {
 ObjectList::const_iterator::const_iterator(const ObjectList* parent, bool end)
 {
     _parent = parent;
-    _subList = _parent->_subLists.size();
+    _subList = end ? _parent->_subLists.size() : 0;
```
Comment (Gymnasiast) [inline src/openrct2/object/ObjectList.cpp:67]: What is this change for?

File: src/openrct2/cmdline/ParkInfoCommands.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    Console::WriteLine("File contains the following objects: ");
+    Console::WriteLine();
+
+    const std::array<std::string, 17> typeToName = {
+        "Ride",        "SmallScenery", "LargeScenery", "Walls",           "Banners",          "Paths",
+        "PathBits",    "SceneryGroup", "ParkEntrance", "Water",           "ScenarioText",     "TerrainSurface",
+        "TerrainEdge", "Station",      "Music",        "FootpathSurface", "FootpathRailings",
+    };
+    const std::array<std::string, 9> sourceGameToName = {
+        "Custom", "WackyWorlds", "TimeTwister", "OpenRCT2Official", "RCT1", "AddedAttractions", "LoopyLandscapes", "", "RCT2",
```
Comment (Gymnasiast) [inline src/openrct2/cmdline/ParkInfoCommands.cpp:97]: Any reason these lack a space?

File: src/openrct2/cmdline/ParkInfoCommands.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        // Save is an S6 (RCT2 format)
+        parkImporter = ParkImporter::CreateS6(objectRepository);
+    }
+    auto loadResult = parkImporter->LoadSavedGame(sourcePath.c_str());
+
+    Console::WriteLine("File contains the following objects: ");
+    Console::WriteLine();
+
+    const std::array<std::string, 17> typeToName = {
+        "Ride",        "SmallScenery", "LargeScenery", "Walls",           "Banners",          "Paths",
```
Comment (Gymnasiast) [inline src/openrct2/cmdline/ParkInfoCommands.cpp:92]: Do we want these in English only?

Comment (duncanspumpkin) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:92]: All the other console commands are English only.

File: src/openrct2/cmdline/ParkInfoCommands.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:97]: It's to align things with the enum. I am sort of designing this for other programs to read so spaces would be annoying.

File: src/openrct2/object/ObjectList.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/object/ObjectList.cpp:67]: I can't remember anymore. I'll have to check

**Comment by duncanspumpkin** [comment]
Interestingly doing this has shown that original id's are overriding identifiers if both are specified.

File: src/openrct2/cmdline/ParkInfoCommands.cpp (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:92]: That isn’t really a good reason. But if this is indeed to be used as input for other programs, then it makes sense to have it in English only.

Comment (IntelOrca) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:92]: These are power user commands, so anyone using them is likely to need to understand English anyway. I could be wrong, but I doubt the time and effort is worth it (not just dev time, but also translators time).

Comment (Gymnasiast) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:92]: I am a power user, but the difference between reading English and Dutch is quite clear - the latter takes significantly less focus and energy.

Comment (duncanspumpkin) [reply src/openrct2/cmdline/ParkInfoCommands.cpp:92]: Are we happy then with the pr overall?

File: src/openrct2/cmdline/ParkInfoCommands.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        "PathBits",    "SceneryGroup", "ParkEntrance", "Water",           "ScenarioText",     "TerrainSurface",
```
Comment (Gymnasiast) [inline src/openrct2/cmdline/ParkInfoCommands.cpp:93]:
```suggestion
        "PathAdditions",    "SceneryGroup", "ParkEntrance", "Water",           "ScenarioText",     "TerrainSurface",
```

**Review by Gymnasiast — COMMENTED** [review]
Only one thing I’d like to see changed, see line note

File: src/openrct2/cmdline/ParkInfoCommands.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2021 OpenRCT2 developers
```
Comment (Broxzier) [inline src/openrct2/cmdline/ParkInfoCommands.cpp:2]:
```suggestion
 * Copyright (c) 2014-2022 OpenRCT2 developers
```

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**